pyresample/geometry.py

```diff
@@ -475,6 +475,57 @@ def __str__(self):
                                                     str(self.lons),
                                                     str(self.lats))
 
+    def geocentric_resolution(self, ellps='WGS84', radius=None, nadir_factor=2):
+        """Calculate maximum geocentric pixel resolution.
+
+        If `lons` is a :class:`xarray.DataArray` object with a `resolution`
+        attribute, this will be used instead of loading the longitude and
+        latitude data. In this case the resolution attribute is assumed to
+        mean the nadir resolution of a swath and will be multiplied by the
+        `nadir_factor` to adjust for increases in the spatial resolution
+        towards the limb of the swath.
+
+        Args:
+            ellps (str): PROJ Ellipsoid for the Cartographic projection
+                used as the target geocentric coordinate reference system.
+                Default: 'WGS84'. Ignored if `radius` is provided.
+            radius (float): Spherical radius of the Earth to use instead of
+                the definitions in `ellps`.
+            nadir_factor (int): Number to multiply the nadir resolution
+                attribute by to reflect pixel size on the limb of the swath.
```

Comment:
For most of the satellites we support, I think this should be more like 5. Needs to be verified though.
Also, could we use the viewing angle to help us find the factor ?

Reply:
At least in pyresample, we don't have the viewing angle. If Satpy wants to override that I am open to that idea.